lib/pbench/test/unit/server/query_apis/test_datasets_detail.py

```diff
@@ -97,38 +63,15 @@ def test_missing_keys(self, client, server_config, keys, user_ok, pbench_token):
         Note that "name", "start", and "end" are required whereas "user" is not mandatory;
         however, Pbench will silently ignore any additional keys that are
         specified.
-        """
-        response = client.post(
-            f"{server_config.rest_uri}/datasets/detail",
-            headers={"Authorization": "Bearer " + pbench_token},
-            json=keys,
-        )
-        assert response.status_code == HTTPStatus.BAD_REQUEST
-        missing = [k for k in ("name", "start", "end") if k not in keys]
-        assert (
-            response.json.get("message")
-            == f"Missing required parameters: {','.join(missing)}"
-        )
-
-    def test_bad_dates(self, client, server_config, user_ok, pbench_token):
-        """
-        Test behavior when a bad date string is given
-        """
-        response = client.post(
-            f"{server_config.rest_uri}/datasets/detail",
-            headers={"Authorization": "Bearer " + pbench_token},
-            json={
-                "user": "drb",
-                "name": "footest",
-                "start": "2020-12",
-                "end": "2020-19",
-            },
-        )
-        assert response.status_code == HTTPStatus.BAD_REQUEST
-        assert (
-            response.json.get("message")
-            == "Value '2020-19' (str) cannot be parsed as a date/time string"
-        )
+       """
+        self.required_keys = [
+            p[0]
+            for p in DatasetsDetail(
+                client.config, client.logger
+            ).schema.parameters.items()
+            if p[1].required
+        ]
```

Comment:
I think it would be nicer if you destructured the tuples in the `for` loop, e.g., something like:
```
        self.required_keys = [
            key
            for key, parameter in DatasetsDetail(
                client.config, client.logger
            ).schema.parameters.items()
            if parameter.required
        ]
```